seatunnel-connectors-v2/connector-kafka/src/main/java/org/apache/seatunnel/connectors/seatunnel/kafka/source/KafkaConsumerThread.java

```diff
@@ -58,8 +61,7 @@ public void run() {
                         task.accept(consumer);
                     }
                 } catch (Exception e) {
-                    throw new KafkaConnectorException(
-                            KafkaConnectorErrorCode.CONSUME_THREAD_RUN_ERROR, e);
```

Comment:
As you described in PR content, before change the code. The exception will be throwed. But you also say it will not throw exception in https://github.com/apache/seatunnel/pull/7273#issuecomment-2268680927. Which is right?

Reply:
https://github.com/apache/seatunnel/blob/16eeb1c12323b462474ad23103de96c0060a4537/seatunnel-connectors-v2/connector-kafka/src/main/java/org/apache/seatunnel/connectors/seatunnel/kafka/source/KafkaConsumerThread.java#L71
The exception is triggered by commit：
https://github.com/apache/seatunnel/blob/16eeb1c12323b462474ad23103de96c0060a4537/seatunnel-connectors-v2/connector-kafka/src/main/java/org/apache/seatunnel/connectors/seatunnel/kafka/source/KafkaSourceReader.java#L289
Poll will not throw an exception：
https://github.com/apache/seatunnel/blob/16eeb1c12323b462474ad23103de96c0060a4537/seatunnel-connectors-v2/connector-kafka/src/main/java/org/apache/seatunnel/connectors/seatunnel/kafka/source/KafkaSourceReader.java#L147

Reply:
I think we should just let job failed when find an exception like https://github.com/apache/seatunnel/blob/16eeb1c12323b462474ad23103de96c0060a4537/seatunnel-connectors-v2/connector-kafka/src/main/java/org/apache/seatunnel/connectors/seatunnel/kafka/source/KafkaConsumerThread.java#L71

In other cases, the behavior not change.

Reply:
The code of conduct is that if we encounter an exception, we terminate the job. If we do not encounter an exception, we do not terminate the job.

Reply:
ok.

Reply:
> I think we should just let job failed when find an exception like
> 
> https://github.com/apache/seatunnel/blob/16eeb1c12323b462474ad23103de96c0060a4537/seatunnel-connectors-v2/connector-kafka/src/main/java/org/apache/seatunnel/connectors/seatunnel/kafka/source/KafkaConsumerThread.java#L71
> 
> In other cases, the behavior not change.

This exception is triggered when consumer.commitSync(offsets). Now there are two problems:
1. This exception is thrown during the execution of the KafkaConsumerThread thread. To obtain the result of the thread execution exception, Future or callback is required. This has a great impact on the current KafkaConsumerThread code structure transformation. I suggest that it keep the thread from exiting abnormally until the Kafka cluster is restored and continue to execute
2. consumer.commitSync is triggered at checkpoint, so even if an exception is captured, the task may not exit abnormally.

Reply:
> To obtain the result of the thread execution exception, Future or callback is required. This has a great impact on the current KafkaConsumerThread code structure transformation.

It's hard but right way. We need do this.

> consumer.commitSync is triggered at checkpoint, so even if an exception is captured, the task may not exit abnormally.

The checkpoint error will let job failed normally.